Create migrating-to-v2.md (#2897)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud-apis/migrating-to-v2.md

---
title: "Migrating to dbt Cloud Administrative API v2"
description: "You should migrate to API v2 while we deprecate API v4 "
sidebar_label: "Migrating to API v2"

> id: "admin-cloud-api"

Author: ```suggestion
id: "migrating-to-v2"
```

---
title: "Migrating to dbt Cloud Administrative API v2"
description: "You should migrate to API v2 while we deprecate API v4 "
sidebar_label: "Migrating to API v2"
id: "migrating-to-v2"
---

> When migrating from dbt Cloud Administrative API v4 to API v2, there are a few differences you should account when querying your data. 

Author: ```suggestion
When migrating from dbt Cloud Administrative API v4 to API v2, there are a few differences you should consider when querying your data. 
```

> When migrating from dbt Cloud Administrative API v4 to API v2, there are a few differences you should consider when querying your data. 

Author: ```suggestion
When migrating from dbt Cloud Administrative API v4 to API v2, there are a few differences you should consider when querying your dbt Cloud account. 
```

## Key differences

> When using the `List runs` endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own:

Author: ```suggestion
When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own:
```

> When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own:

Author: ```suggestion
When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `cloud.getdbt.com` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan. For example, if your region is EMEA multi-tenant, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id`:
```

```shell

> GET https://cloud.getdbt.com/api/v2/accounts/{accountId}/runs/?include_related=["trigger"]&order_by=-id

Author: ```suggestion
GET https://cloud.getdbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id
```

> GET https://cloud.getdbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id

Reviewer: One minor nit, do we have precedent yet for handling URLs where we need to reference emea/au/us separately?

Reviewer 2: i _think_ Leona (who's OOO) has created this placeholder text that would follow the example url:

> Replace `https://cloud.getdbt.com` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan. For example, if your region is EMEA multi-tenant, your endpoint is `GET https://emea.dbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id`

another example [here](https://docs.getdbt.com/docs/dbt-cloud-apis/metadata-api#browse-the-api)

Author: I've addressed this! Thanks for bringing it up.

```

Differences in responses include:

> | Property | API v4    | API v2     |

Author: ```suggestion
| Property description | API v4    | API v2     |
```

> |---------|-----------|-------------|

Author: ```suggestion
|---------------------|-----------|-------------|
```

> When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `cloud.getdbt.com` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan. For example, if your region is EMEA multi-tenant, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id`:

Author: ```suggestion
When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `{YOUR_ACCESS_URL}` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan. For example, if your region is EMEA multi-tenant, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id`:
```

> When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `{YOUR_ACCESS_URL}` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan. For example, if your region is EMEA multi-tenant, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id`:

Author: ```suggestion
When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `{YOUR_ACCESS_URL}` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan.
```

> | Use sort by `-id` to reverse the sort order | Defaults to order by most recent | Defaults to order by least recent |

Author: ```suggestion
| Reverse sort order when you use sort by `-id`  | Defaults to order by most recent | Defaults to order by least recent |
```

> | timestamps | Unix timestamps | ISO strings |

Author: ```suggestion
| Timestamps | Unix timestamps | ISO strings |
```

> | Timestamps | Unix timestamps | ISO strings |

Author: ```suggestion
| Change timestamps | Unix timestamps | ISO strings |
```

> GET https://cloud.getdbt.com/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id

Author: ```suggestion
GET https://{YOUR_ACCESS_URL}/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id
```

> | Change timestamps | Unix timestamps | ISO strings |

Author: ```suggestion
| Update to timestamps | Unix timestamps | ISO strings |
```

> 

Author: For example, if your region is EMEA multi-tenant and your account ID is `001`, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/001/runs/?include_related=[%22trigger%22]&order_by=-id`

> 

Author: ```suggestion

 For example, if your region is EMEA multi-tenant and your account ID is `001`, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/001/runs/?include_related=[%22trigger%22]&order_by=-id`
```

> | `id`, `environment_id`, `account_id`, `project_id`, `job_id` | Values are the same, but they are strings | Values are the same, but they are numeric |

Author: ```suggestion
| Updated IDs: `id`, `environment_id`, `account_id`, `project_id`, `job_id` | Values are the same, but they are strings | Values are the same, but they are numeric |
```

> | Updated IDs: `id`, `environment_id`, `account_id`, `project_id`, `job_id` | Values are the same, but they are strings | Values are the same, but they are numeric |

Author: ```suggestion
| Update to IDs: `id`, `environment_id`, `account_id`, `project_id`, `job_id` | Values are the same, but they are strings | Values are the same, but they are numeric |
```

> | Status | `status` property |  Maps to `status_humanized` |

Author: ```suggestion
| Returns runs with the specified status | `status` property |  Maps to `status_humanized` |
```

> | Returns runs with the specified status | `status` property |  Maps to `status_humanized` |

Author: ```suggestion
| Return runs with the specified status | `status` property |  Maps to `status_humanized` |
```

> | Replace | `replace` property | Maps to the `trigger` property |

Author: ```suggestion
| Related field included with run | `replace` property | Maps to the `trigger` property |
```

> | Return runs with the specified status | `status` property |  Maps to `status_humanized` |

Author: ```suggestion
| New property for returning runs with the specified status | `status` property |  Maps to `status_humanized` |
```

> | Related field included with run | `replace` property | Maps to the `trigger` property |

Author: ```suggestion
| Include related field with run | `replace` property | Maps to the `trigger` property |
```

> | Include related field with run | `replace` property | Maps to the `trigger` property |

Author: ```suggestion
| New property for including related field with run | `replace` property | Maps to the `trigger` property |
```

---
title: "Migrating to dbt Cloud Administrative API v2"
description: "You should migrate to API v2 while we deprecate API v4 "
sidebar_label: "Migrating to API v2"
id: "migrating-to-v2"
---

When migrating from dbt Cloud Administrative API v4 to API v2, there are a few differences you should consider when querying your dbt Cloud account. 

## Key differences

When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `{YOUR_ACCESS_URL}` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan.

```shell
GET https://{YOUR_ACCESS_URL}/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id
```

>  For example, if your region is EMEA multi-tenant and your account ID is `001`, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/001/runs/?include_related=[%22trigger%22]&order_by=-id`

Author: ```suggestion
For example, if your region is EMEA multi-tenant and your account ID is `001`, your endpoint would be  `GET https://emea.dbt.com/api/v2/accounts/001/runs/?include_related=[%22trigger%22]&order_by=-id`
```

---
title: "Migrating to dbt Cloud Administrative API v2"
description: "You should migrate to API v2 while we deprecate API v4 "
sidebar_label: "Migrating to API v2"
id: "migrating-to-v2"
---

> 

Reviewer 3: Should we include a blurb here about why we're deprecating V4 and asking them to step down to V2 (maybe even why there is no V3)? Security teams might see reverting to a lower version as a red flag.

In an attempt to provide an improved dbt Cloud Administrative API experience, the dbt Cloud API v4 will be deprecated by April 30th, 2023. We suggest you migrate to dbt Cloud Administrative API v2. When migrating from API v4 to API v2, there are a few differences you should consider when querying your dbt Cloud account.

## Key differences

When using the [List runs](/dbt-cloud/api-v2#tag/Runs) endpoint, you can include triggered runs and sort by ID. You can use the following request in v2 to get a similar response as v4, replacing the `{accountId}` with your own and `{YOUR_ACCESS_URL}` with the appropriate [Access URL](https://docs.getdbt.com/docs/deploy/regions-ip-addresses) for your region and plan:

```shell
GET https://{YOUR_ACCESS_URL}/api/v2/accounts/{accountId}/runs/?include_related=[%22trigger%22]&order_by=-id
```

> 

Author: ```suggestion
```
